src/sst/core/eli/elibase.cc

```diff
@@ -20,32 +20,16 @@ namespace SST {
 **************************************************************************/
 namespace ELI {
 
-std::unique_ptr<LoadedLibraries::LibraryMap> LoadedLibraries::loaders_ {};
-
 bool
 LoadedLibraries::addLoader(
     const std::string& lib, const std::string& name, const std::string& alias, LibraryLoader* loader)
 {
-    if ( !loaders_ ) { loaders_ = std::unique_ptr<LibraryMap>(new LibraryMap); }
-    (*loaders_)[lib][name].push_back(loader);
-    if ( !alias.empty() ) (*loaders_)[lib][alias].push_back(loader);
-    return true;
-}
+    std::shared_ptr<LibraryLoader> shared_loader(loader);
 
-const LoadedLibraries::LibraryMap&
-LoadedLibraries::getLoaders()
-{
-    if ( !loaders_ ) { loaders_ = std::unique_ptr<LibraryMap>(new LibraryMap); }
-    return *loaders_;
-}
-
-bool
-LoadedLibraries::isLoaded(const std::string& name)
-{
-    if ( loaders_ ) { return loaders_->find(name) != loaders_->end(); }
-    else {
-        return false; // nothing loaded yet
-    }
+    auto& library = getLoaders()[lib];
+    if ( !alias.empty() && alias != name ) library[alias].push_back(shared_loader);
+    library[name].push_back(std::move(shared_loader));
```

Comment:
On the last use of `shared_loader` we use `std::move()` so that it doesn't have the overhead of incrementing the reference count during the `push_back()` and then decrementing the reference count when `shared_loader` is destroyed. We move `shared_loader` to the new entry without changing reference counts, and its destruction is a no-op.